bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/esm_nv.py

```diff
@@ -213,22 +214,38 @@ def forward(
                 if self.config.attn_input_format == "bshd":
                     te_rope_emb = self.rotary_embeddings(max_seq_len=hidden_states.shape[1])
                 elif self.config.attn_input_format == "thd":
-                    te_rope_emb = self.rotary_embeddings(max_seq_len=kwargs["cu_seq_lens_q"][-1])
+                    if self.config.use_cp:
+                        te_rope_emb = self.rotary_embeddings(max_seq_len=kwargs["cu_seq_lens_q_padded"][-1])
+                    else:
+                        te_rope_emb = self.rotary_embeddings(max_seq_len=kwargs["cu_seq_lens_q"][-1])
```

Comment:
id just check for a `cu_seq_lens_q_padded` and use `cu_seq_lens_q` of its not there